wikiclass/feature_lists/wikidatawiki.py

```diff
@@ -21,6 +32,164 @@ class items:
+				try:
+					if(list_sources_in_JSON_content['numeric-id'] == int(line[0])): #if a source comes from Wikimedia projects and DBpedia (i.e. DBpedia collects data from Wikipedia), append that source to the list
+						list_wikimedia_sources.append(list_sources_in_JSON_content)
+						break
+				except:
+					continue
+	return list_wikimedia_sources
+
+def _process_external_sources(item):
+    
+    count_all_sources = len(_process_all_sources(item))
+    count_wiki_sources = len(_process_wikimedia_sources(item))
+    return count_all_sources - count_wiki_sources
+
+
+def _process_external_sources_ratio(item):
+	
+	try: 
+		count_external_sources = _process_external_sources(item)
+		count_claims_with_sources = len(_process_sources(item))
+		return count_external_sources/count_claims_with_sources
+	except: 
+		return 0.0 #if the revision does not contain any sources, return 0 
+	
+def _process_unique_sources(item):
+	
+	list_sources_in_JSON = []
+	result_set = set()
+	
+	for property in item.claims:
+		list_of_properties = property
+		for claim in item.claims[list_of_properties]:
+			list_of_claims = claim
+			for i, source in enumerate(list_of_claims.sources):
+				for index_list_1, index_list_2 in source.items(): 
+					sources_in_JSON = index_list_2[0].toJSON()
+					list_sources_in_JSON.append("property : " + str(sources_in_JSON['property']) + " & value : " + str(sources_in_JSON['datavalue']['value']))
+	
+	#eliminate duplicates
+	for value in list_sources_in_JSON:
+		result_set.add(value)
+	
+	return result_set
+	
+
+def _process_complete_translations(item):
+	item_labels_dict = item.labels
+	item_desc_dict = item.descriptions
+	combined_dict = {}
+	result_set = []
+	
+	#merging item label dictionary and item description dictionary
+	for key in (item_labels_dict.keys() | item_desc_dict.keys()):
+		if key in item_labels_dict: combined_dict.setdefault(key, []).append(item_labels_dict[key])
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if a language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if(len(value[1]) == 2): 
+			result_set.append(value[0])
+			
+	return result_set
+
+def _process_important_translations(item):
+	item_labels_dict = item.labels
+	item_desc_dict = item.descriptions
+	combined_dict = {}
+	result_set = []
+	
+	#merging item label dictionary and item description dictionary
+	for key in (item_labels_dict.keys() | item_desc_dict.keys()):
+		if key in item_labels_dict: combined_dict.setdefault(key, []).append(item_labels_dict[key])
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if an important language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if((len(value[1]) == 2) and (value[0] in ('en', 'de', 'ar', 'zh', 'es', 'pt', 'ru', 'fr'))):
```

Comment:
These lang codes should be in a top level variable `IMPORTANT_LANG_CODES` with a link to the item quality page section in a comment.